lib/pengine/complex.c

```diff
@@ -182,10 +190,10 @@ get_meta_attributes(GHashTable * meta_hash, pcmk_resource_t * rsc,
     }
 
     for (xmlAttrPtr a = pcmk__xe_first_attr(rsc->xml); a != NULL; a = a->next) {
-        const char *prop_name = (const char *) a->name;
-        const char *prop_value = pcmk__xml_attr_value(a);
-
-        add_hash_param(meta_hash, prop_name, prop_value);
+        if (a->children != NULL) {
+            dup_attr((gpointer) a->name, (gpointer) a->children->content,
```

Comment:
Why did we decide to use `dup_attr()` here (vs. `pcmk__insert_meta()`)? The choice in `expand_parents_fixed_nvpairs()` is clearer

Reply:
`pcmk__insert_meta()` inserts into `(obj)->meta`, whereas here we have the table that we need to insert into.

To override or not doesn't matter here as long as we pass an empty hash table to `get_meta_attributes()`. I just noticed there is one caller that does not: `register_if_fencing_device()` unpacks a resource's meta-attributes a second time passing a node for rule evaluation. The evaluated values won't override any existing non-rule-based values. That's probably a bug.